src/ringmesh/io/geomodel/io_adeli.cpp

```diff
@@ -76,125 +304,78 @@ namespace
                         "I/O", "Adeli supports only tet meshes" );
                 }
             }
-
-            write_vertices( geomodel_mesh, out );
-
-            write_mesh_elements( geomodel, out );
+            write_regions_vertices( geomodel, out );
+            write_regions( geomodel, out );
             out << std::flush;
         }
 
     private:
-        void write_vertices(
-            const GeoModelMesh3D& geomodel_mesh, std::ofstream& out ) const
-        {
-            out << "$NOD" << EOL;
-            out << geomodel_mesh.vertices.nb() << EOL;
-            for( auto v : range( geomodel_mesh.vertices.nb() ) )
-            {
-                out << v + id_offset_adeli << " "
-                    << geomodel_mesh.vertices.vertex( v ) << EOL;
-            }
-            out << "$ENDNOD" << EOL;
-        }
-
-        index_t write_corners(
-            const GeoModel3D& geomodel, std::ofstream& out ) const
+        void write_regions( const GeoModel3D& geomodel, std::ofstream& out )
         {
             out << "$ELM" << EOL;
-            out << nb_total_elements( geomodel ) << EOL;
-            index_t elt = 1;
-            for( const auto& corner : geomodel.corners() )
+            out << count_regions_and_deps_elements( geomodel ) << EOL;
+            std::vector< RegionAndDependentEntities > regions_and_deps;
+            regions_and_deps.emplace_back( geomodel,
+                geomodel.region( 0 ).gmme(), id_offset_adeli, id_offset_adeli,
+                id_offset_adeli, id_offset_adeli, id_offset_adeli,
+                id_offset_adeli );
+            for( index_t region_index = 1; region_index < geomodel.nb_regions();
```

Comment:
Could you prefer range than old loop syntax?

Reply:
I prefrer the old way here because it stats at 1

Reply:
Why not using range( 1, nb_regions )?